apps/client/src/features/app-settings/panel/project-panel/ProjectListItem.tsx

```diff
@@ -0,0 +1,165 @@
+import { Menu, MenuButton, IconButton, MenuList, MenuItem } from '@chakra-ui/react';
+import { IoEllipsisHorizontal } from '@react-icons/all-files/io5/IoEllipsisHorizontal';
+import { useEffect, useMemo, useState } from 'react';
+import { AxiosError } from 'axios';
+
+import style from './ProjectPanel.module.scss';
+import { renameProject, loadProject, duplicateProject } from '../../../../common/api/ontimeApi';
+import { ontimeQueryClient } from '../../../../common/queryClient';
+import { PROJECT_LIST } from '../../../../common/api/apiConstants';
+import { EditMode } from './ProjectList';
+import RenameProjectForm, { RenameProjectFormValues } from './RenameProjectForm';
+import DuplicateProjectForm, { DuplicateProjectFormValues } from './DuplicateProjectForm';
+
+interface ProjectListItemProps {
+  current?: boolean;
+  filename: string;
+  createdAt: string;
+  updatedAt: string;
+  onToggleEditMode?: (editMode: EditMode, filename: string | null) => void;
+  onSubmit?: () => void;
+  editingFilename: string | null;
+  editingMode: EditMode | null;
+}
+
+export default function ProjectListItem({
+  current,
+  createdAt,
+  editingFilename,
+  editingMode,
+  filename,
+  onSubmit,
+  onToggleEditMode,
+  updatedAt,
+}: ProjectListItemProps) {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+
+  const handleRefetch = async () => {
+    await ontimeQueryClient.invalidateQueries({ queryKey: PROJECT_LIST });
+  };
+
+  const handleSubmitRename = async (values: RenameProjectFormValues) => {
+    try {
+      if (!values.filename) {
+        setSubmitError('Filename cannot be blank');
+        return;
+      }
+      await renameProject(filename, values.filename);
+      await handleRefetch();
+      onSubmit?.();
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        const errorMessage = error?.response?.data?.message;
+
+        setSubmitError(errorMessage);
+      } else {
+        setSubmitError('An unknown error occurred');
+      }
+    }
+  };
+
+  const handleSubmitDuplicate = async (values: DuplicateProjectFormValues) => {
+    try {
+      if (!values.newFilename) {
+        setSubmitError('Filename cannot be blank');
+        return;
+      }
+      await duplicateProject(filename, values.newFilename);
+      await handleRefetch();
+      onSubmit?.();
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        const errorMessage = error?.response?.data?.message;
+
+        setSubmitError(errorMessage);
+      } else {
+        setSubmitError('An unknown error occurred');
+      }
+    }
+  };
+
+  const handleCancel = () => {
+    onToggleEditMode?.(null, null);
+  };
+
+  useEffect(() => {
+    setSubmitError(null);
+  }, [editingMode]);
```

Comment:
can you please double-check this? we likely want to change this in a callback.
In my experience, leveraging `useEffect` this way is more often than not a source for bugs